signal_opt_service/src/signal_opt_processing_worker.cpp

```
@@ -1,17 +1,61 @@
 #include "signal_opt_processing_worker.h"
 namespace signal_opt_service
 {
+    
+    void signal_opt_processing_worker::configure_dpp_optimizer(
+                    const streets_signal_optimization::streets_desired_phase_plan_generator_configuration &dpp_config) {
+        dpp_arbitrator_ptr = std::make_shared<streets_signal_optimization::streets_desired_phase_plan_arbitrator>();
+        dpp_generator_ptr = std::make_shared<streets_signal_optimization::streets_desired_phase_plan_generator>();
+        dpp_generator_ptr->set_configuration(dpp_config.initial_green_buffer,
+                                            dpp_config.final_green_buffer,
+                                            dpp_config.et_inaccuracy_buffer,
+                                            dpp_config.queue_max_time_headway,
+                                            dpp_config.so_radius,
+                                            dpp_config.min_green,
+                                            dpp_config.max_green,
+                                            dpp_config.desired_future_move_group_count);
+    }
+    
+    streets_desired_phase_plan::streets_desired_phase_plan signal_opt_processing_worker::convert_spat_to_dpp(
+                        const std::shared_ptr<signal_phase_and_timing::spat> &spat_ptr,
+                        const std::shared_ptr<streets_signal_optimization::movement_groups> &move_groups) const
+    {
+        streets_desired_phase_plan::streets_desired_phase_plan spat_dpp;
+    
+        signal_phase_and_timing::intersection_state int_state = spat_ptr->get_intersection();
+        if (dpp_generator_ptr) {
+            spat_dpp = dpp_generator_ptr->convert_spat_to_dpp(int_state, move_groups);
+        }
+        else {
+            SPDLOG_DEBUG("dpp_generator_ptr is not initialized!");
```

If the dpp generator is not initialized should we be throwing and exception or at least logging something higher than info. The method that calls this, would it fail if given an empty desired phase plan. If so, we may as well throw an exception here.

---

So, the library will throw an exception if the desired_phase_plan is empty. But I agree that here we should also throw an exception and do not continue. I have added a throw statement here and will catch it in signal_opt_service.